### PR TITLE
Small bug fix and update package limits

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -12,8 +12,8 @@ Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 
 [compat]
 DataFrames = "≥ 0.17.1"
-LinearOperators = "≥ 0.4.0"
-NLPModels = "0.7.0, 0.8.0, 0.9"
+LinearOperators = "0.4.0, 0.5.0, 0.6.0, 0.7.0"
```
Comment (dpo) [inline Project.toml]: Why list the versions instead of specifying a range?

File: test/test_bmark.jl [hunk 2]
```diff
@@ -13,8 +13,8 @@ function test_bmark()
     callable = CallableSolver()
     stats = solve_problems(dummy_solver, problems)
     @test stats isa DataFrame
-    stats = solve_problems(dummy_solver, problems, reset_solver=false)
-    stats = solve_problems(dummy_solver, problems, reset_solver=true)
+    stats = solve_problems(dummy_solver, problems, reset_problem=false)
+    stats = solve_problems(dummy_solver, problems, reset_problem=true)
```
Comment (dpo) [inline test/test_bmark.jl]: How was this passing before?

Comment (abelsiqueira) [reply test/test_bmark.jl]: `solve_problems` accept `kwargs...` to pass to the solver inside a `try` block, so it passed the tests even though it failed.

File: Project.toml (hunk 1 above)
Comment (abelsiqueira) [reply Project.toml]: Looks like it doesn't work yet: https://discourse.julialang.org/t/explicit-version-ranges-in-compat-section-in-project-toml/30147

Comment (dpo) [reply Project.toml]: OK.

**Pull request merged**